This file is part of MXE. See LICENSE.md for licensing information.

Contains ad hoc patches for cross building.

From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Marco Borrini <borrinimarco74@gmail.com>
Date: Tue, 4 Aug 2026 16:33:00 +0200
Subject: [PATCH 1/1] Add libstdc++ and Windows libraries to pkg-config private libs

When building OpenAL as a static library with a C++ compiler, consumers
that link it using a C compiler via pkg-config (e.g., test-openal.c in MXE)
need to link against the C++ standard library, otherwise they will fail
with undefined references to std:: symbols and __cxa exceptions.

In addition, standard Windows libraries (ole32, uuid, shell32) are required
by OpenAL internally, but are not explicitly injected by CMake into
pkg-config when compiling a static library, because CMake assumes it will
always handle linking when consumers use CMake instead of pkg-config.

diff -ur openal-soft-1.25.2-orig/CMakeLists.txt openal-soft-1.25.2-new/CMakeLists.txt
--- openal-soft-1.25.2-orig/CMakeLists.txt	2026-08-04 16:33:39.319656642 +0200
+++ openal-soft-1.25.2-new/CMakeLists.txt	2026-08-04 16:33:47.363418526 +0200
@@ -1559,6 +1559,12 @@
             set(PKG_CONFIG_PRIVATE_LIBS "${PKG_CONFIG_PRIVATE_LIBS} -l${FLAG}")
         endif()
     endforeach()
+    if(CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang")
+        set(PKG_CONFIG_PRIVATE_LIBS "${PKG_CONFIG_PRIVATE_LIBS} -lstdc++")
+    endif()
+    if(WIN32)
+        set(PKG_CONFIG_PRIVATE_LIBS "${PKG_CONFIG_PRIVATE_LIBS} -lole32 -luuid -lshell32")
+    endif()
 endif()
 
 if(UNIX_ELF)
